WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostViewModel.kt

```diff
@@ -0,0 +1,183 @@
+        updatedTitle: String,
+        getUpdatedContent: ((String) -> String?)
+    ): UpdateResult {
+        if (!mEditPostRepository.hasPost()) {
+            AppLog.e(AppLog.T.POSTS, "Attempted to save an invalid Post.")
+            return Error
+        }
+        return mEditPostRepository.updateInTransaction { postModel ->
+            getUpdatedContent(postModel.content)?.let { updatedContent ->
+                val postTitleOrContentChanged = updatePostContentNewEditor(
+                        context,
+                        mShowAztecEditor,
+                        mEditPostRepository,
+                        postModel,
+                        updatedTitle,
+                        updatedContent
+                )
+
+                // only makes sense to change the publish date and locally changed date if the Post was actually changed
+                if (postTitleOrContentChanged) {
+                    mEditPostRepository.updatePublishDateIfShouldBePublishedImmediately(postModel)
+                    val currentTime = localeManagerWrapper.getCurrentCalendar()
+                    postModel
+                            .setDateLocallyChanged(
+                                    dateTimeUtils.iso8601FromCalendar(currentTime)
+                            )
+                }
+
+                Success(postTitleOrContentChanged)
+            } ?: Error
+        }
+    }
+
+    /**
+     * Updates post object with given title and content
+     */
+    private fun updatePostContentNewEditor(
+        context: Context,
+        showAztecEditor: Boolean,
+        postRepository: EditPostRepository,
+        editedPost: PostModel?,
+        title: String,
+        content: String
+    ): Boolean {
+        if (editedPost == null) {
+            return false
+        }
+        val titleChanged = editedPost.title != title
+        editedPost.setTitle(title)
+        val contentChanged: Boolean
+        when {
+            mediaInsertedOnCreation -> {
+                mediaInsertedOnCreation = false
+                contentChanged = true
+            }
+            isCurrentMediaMarkedUploadingDifferentToOriginal(
+                    context,
+                    showAztecEditor,
+                    editedPost.content
+            ) -> contentChanged = true
+            else -> contentChanged = editedPost.content.compareTo(content) != 0
+        }
+        if (contentChanged) {
+            editedPost.setContent(content)
+        }
+
+        val statusChanged = postRepository.hasStatusChanged(editedPost.status)
+
+        if (!editedPost.isLocalDraft && (titleChanged || contentChanged || statusChanged)) {
+            editedPost.setIsLocallyChanged(true)
+            val currentTime = localeManagerWrapper.getCurrentCalendar()
+            editedPost
+                    .setDateLocallyChanged(dateTimeUtils.iso8601FromCalendar(currentTime))
+        }
+
+        return titleChanged || contentChanged
```

Comment:
Shouldn't we add `|| statusChanged` here?

Reply:
This follows the previous logic. My guess would be (from the method name - `updatePostContentNewEditor`  that we only want to return true if the post content has changed (and by content we mean the 2 fields that are editable - title and content).

Reply:
Since it seems like a bug I think we might want to consider fixing it - either stop updating the status or renaming the method and returning true when the status changes. However, let's not do it in this PR.